sympy/functions/combinatorial/factorials.py

```diff
@@ -346,6 +346,8 @@ def eval(cls, arg):
                 raise ValueError("argument must be nonnegative or odd")
 
             return arg * (S.NegativeOne) ** ((1 - arg) / 2) / factorial2(-arg)
+        else:
```

Comment:
If arg.is_integer is None, it shouldn't give an error. We don't want to require people to set `x = Symbol('x', integer=True)` just to write `factorial2(x)`.